source/_posts/marionette-v3.md

```diff
@@ -0,0 +1,243 @@
+title: Marionette.js v3. Getting started with new version 
+date: 2016-08-10 21:19:00 
+tags: 
+- backbone
+- marionette
+- learning
+- v3
+
+---
+
+New version of `Marionette.js` framework was released.
+
+Welcome to the party
+for people who know this cool framework and
+for someone who are not so much.
+We will go through all new changes which v3 brought us.
+
+
+## View
+
+Version 2.x had a lot of different kind of views: `View`, `ItemView`,
+`LayoutView`, `CollectionView`, `CompositeView`.
+
+In version 3 `View`, `ItemView`, `LayoutView` were 'merged' into `View`,
+`CompositeView` became deprecated, so this list of views was reduced and
+now we have just `View` and `CollectionView`.
+
+Here some simple examples:
+
+````js
+import Mn from 'backbone.marionette';
+
+const template = _.template('<h1>Marionette says hello!</h1>');
+const MyView = Mn.View.extend({
+  el: 'body',
+  template: template
+});
+const myView = new MyView();
+
+myView.render();
+````
+
+[Live example](https://jsfiddle.net/599dtsb7/24/)
+
+
+#### How does `View` handle regions?
+The same how `LayoutView` did.
+
+````js
+import Mn from 'backbone.marionette';
+
+const template1 = _.template('<h1>Marionette says hello!</h1>');
+const template2 = _.template('<h1>Marionette is awesome!</h1>');
+
+const myView1 = new Mn.View({template: template1});
+const myView2 = new Mn.View({template: template2});
+
+const MyView = Mn.View.extend({
+  el: '#container',
+  template: false,
+  regions: {
+  	region1: '#region1',
+    region2: '#region2'
+  },
+
+  onRender() {
+  	this.showChildView('region1', myView1);
+    this.showChildView('region2', myView2)
+  }
+});
+const myView = new MyView();
+
+myView.render();
+````
+
+[Live example](https://jsfiddle.net/599dtsb7/26/)
+
+
+#### How can I build table or tree without `CompositeView`?
+
+If you build your table or tree using `CompositeView` just take a look [here](http://marionettejs.com/docs/v3.0.0-pre.5/marionette.collectionview.html#rendering-tables)
+how easy migrate on using `View` and `CollectionView`.
+
+
+### View events
+
+Since `version 3` there are no `show`, `before:show` events on the view,
+so you can use on `render`, `before:render` events.
```

Comment:
> Since `version 3`, `show` and `before:show` events were removed in favor of `render`, `before:render`, or other view events.

might come off more formal~ish